package.json

```diff
@@ -11,6 +11,7 @@
     "url": "https://github.com/ampproject/amphtml.git"
   },
   "scripts": {
+    "filesize": "filesize -c=build-system/tasks/bundle-size/filesize.json",
```

Comment:
_(Repeating what we were discussing via chat)_

I can think of three reasons for not wanting to add this to `package.json`:

1. This functionality is mostly available in `gulp bundle-size --on_local_build`. (See below for a new version with brotli / uncompressed / ratio in table form.)
2. We used to have `npm` scripts (mostly aliases for `gulp` tasks), but they frequently became outdated and added a constant maintenance burden. 
https://github.com/ampproject/amphtml/blob/ee2dea74e954ceaf5c284bfbfb2f81416858fa9a/package.json#L17-L26
3. Since then, our entire toolchain was reorganized in `build-system/tasks/`, with the command `gulp --tasks` serving as always-updated documentation for the entire toolchain.
![image](https://user-images.githubusercontent.com/26553114/107102767-7a30ee00-67e9-11eb-9f80-673725d1eca4.png)
4. (Cosmetic nit) The text output of the `filesize` command uses colors that are hard to read on consoles with dark backgrounds.
![image](https://user-images.githubusercontent.com/26553114/107102574-e5c68b80-67e8-11eb-84e3-04abe50eed3a.png)

Of course, there is one reason for wanting to do this: Developers should be allowed to use whatever works for them, and the maintenance burden is worth it 😃 

Would it work if `gulp bundle-size --on_local_build` were modified as follows for use during local development and CI?

![image](https://user-images.githubusercontent.com/26553114/107102546-c62f6300-67e8-11eb-9655-e192df5fd36e.png)

If yes, I've created #32479 with the changes. If no, I'm good with merging this PR.

Reply:
On CLI styling: It's a toss-up, but I believe the output from the `filesize` CLI is superior. This particular CLI is under active development, and used by more than the amphtml project.

On gulp versus npm scripts: My preference would be to move away from gulp for local developer tasks, and be more similar to other projects in the open source JS Space. 

On the proposed change to the local build gulp command, I don't think we should add additional logic in the amphtml project when already provided by the dependency.

Reply:
**On CLI styling / logging change:** Thanks for setting me straight. I looked through the `filesize` source code and am convinced its output is superior. (And very clever!) I've sent you two PRs that together replace the custom logging in `bundle-size` with output from `filesize`: https://github.com/ampproject/filesize/pull/127 and https://github.com/ampproject/amphtml/pull/32479.

**On gulp vs. npm scripts:** This is the only thing on which I'll offer a little pushback. Several of our tasks rely on `gulp` features (e.g. file streaming, task composition, flag / task listing) for performance gains and usability benefits. While technically possible to rewrite everything without `gulp`, but I'm not sure if it's a) worth the effort, and b) higher priority than other projects. Alternatively, we could add wrappers for all `gulp` tasks to the `scripts` section of `package.json`, but I'm concerned that will re-create the old maintenance problem and contribute to bloat.

**New idea:** What if we adopted something like [`nps`](https://www.npmjs.com/package/nps) to be used for things not covered by `gulp` tasks, and store all its script definitions in a separate file outside `package.json`? This will eliminate duplication of `gulp` tasks, and prevent bloat in `package.json`. I prototyped this in https://github.com/ampproject/amphtml/pull/32481. WDYT?

Reply:
For clarity: This doesn't come across as difficult, you're quite patient with me!

We do disagree about the value of `gulp`, but that doesn't mean we can't move forward and find a good answer.

I'll take a look at the other PRs you opened this weekend on the topic and we can choose one to move forward with.

Reply:
Thanks for the ideas and the reviews. I'm sure there's some middle ground we can arrive at on our use of `gulp` that gets rid of inefficiencies / useless code paths while not forcing us to rewrite all our tasks from scratch. Will definitely follow up with you on this.